src/S3/S3Client.php

```diff
@@ -467,43 +501,81 @@ public function copy(
         $acl = 'private',
         array $options = []
     ) {
+        return $this
+            ->copyAsync($fromBucket, $fromKey, $bucket, $key, $acl, $options)
+            ->wait();
+    }
+
+    /**
+     * Copy an object of any size to a different location asynchronously.
+     *
+     * @param string $fromBucket    Bucket where the copy source resides.
+     * @param string $fromKey       Key of the copy source.
+     * @param string $bucket        Bucket to which to copy the object.
+     * @param string $key           Key to which to copy the object.
+     * @param string $acl           ACL to apply to the copy (default: private).
+     * @param array  $options       Options used to configure the upload process.
+     *
+     * @see self::copy for more info about the parameters above.
+     * @return PromiseInterface     Returns a promise that will be fulfilled
+     *                              with the result of the copy.
+     */
+    public function copyAsync(
+        $fromBucket,
+        $fromKey,
+        $bucket,
+        $key,
```

Comment:
Can you rename these two parameters, bucket and key, to something like `$toBucket`, `$toKey` or `$destBucket`, `$destKey`?